core/src/main/java/kafka/log/remote/CustomMetadataSizeLimitExceededException.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.log.remote;
+
+class CustomMetadataSizeLimitExceededException extends Exception {
```

Comment:
OK, it's internal used only. Should we put under `storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/` like `RemoteStorageException` did?

Reply:
I'm not sure. As you said, it's internal, supposed to be used inside `RemoteLogManager.java` only. Am I missing something?

Reply:
I mean, we can put all remote storage related exception classes under the same package. In `storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/`, there's already a `RemoteStorageException`, we can put `CustomMetadataSizeLimitExceededException` under the same place. WDYT?

Reply:
`storage.api` is a subproject for exposing public stuff. `CustomMetadataSizeLimitExceededException` is an implementation detail of `RemoteLogManager`. `RemoteStorageException` is different because it's also a public exception. That's why I'm a bit unsure if we should expose it by putting in  `storage.api`.

Reply:
OK, make sense to me. Let's keep it as is. Thanks.